src/CameraGroup.hpp

```diff
@@ -35,7 +35,7 @@ class CameraGroup {
   std::vector<int> vis_object_idx_; // vector of index of the 3D object
 
   // extrinsic
-  std::map<int, double *>
+  std::map<int, std::vector<double>>
```

Comment:
oh? it does not cause any problem to have it declared as a vector? I used the * declaration because it is what I have always used for ceres

Reply:
ceres's `AddResidualBlock()` function only wants `double *`, so you can get that from `std::vector<double> v` as `v.data()`. The problem with `double *` before this PR was that when you set it in `CameraGroup::setCameraPoseMat`(and similar functions) `relative_camera_pose_[id_cam] = new double[6];` this would leak the memory as the pointer to this double[6] is lost once you go out of the function scope. Using `std::vector` (or other similar std data structure) solves this problem. Alternatively, you have to clean the memory allocated on heap manually (i.e. `delete []`) but within a proper scope.